crates/wasi-http/wit/deps/http/types.wit

```diff
@@ -185,21 +173,21 @@ interface types {
     /// Get the Method for the Request.
     method: func() -> method;
     /// Set the Method for the Request.
-    set-method: func(method: method);
+    set-method: func(method: method) -> result;
```

Comment:
Only feedback on this PR is that the wit docs here should describe that the set fails iff the value given is syntactically invalid.